src/edu/pugetsound/mathcs/architecture_nlp/features/spf/SPFSemanticAnalyzer.java

```diff
@@ -0,0 +1,61 @@
+package edu.pugetsound.mathcs.architecture_nlp.features.spf;
+
+import edu.cornell.cs.nlp.spf.base.string.IStringFilter;
+import edu.cornell.cs.nlp.spf.base.string.StubStringFilter;
+import edu.cornell.cs.nlp.spf.ccg.lexicon.LexicalEntry;
+import edu.cornell.cs.nlp.spf.data.sentence.Sentence;
+import edu.cornell.cs.nlp.spf.mr.lambda.LogicalExpression;
+import edu.cornell.cs.nlp.spf.parser.IDerivation;
+import edu.pugetsound.mathcs.nlp.architecture_nlp.features.SemanticAnalyzer;
+import edu.pugetsound.mathcs.nlp.lang.Conversation;
+import edu.pugetsound.mathcs.nlp.lang.Utterance;
+
+import java.util.LinkedHashSet;
+import java.util.List;
+/**
+ * This class acts as a platform to take user utterances and derive Logical representations
+ * of the meaning of the utterance. This is done so via the usage of Yoav Artzi's SPF model, 
+ * contained within Interactor.
+ * 
+ * @author Jared Polonitza
+ * @version 1.0
+ */
+public class SPFSemanticAnalyzer implements SemanticAnalyzer{
```

Comment:
Go through and add Javadoc comments for the remaining public methods in this class.